### PR TITLE
caveats: non-macOS platforms warn on services

### LINKED ISSUE #5847: linuxbrew does not support services
**Please note that we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

# What you were trying to do (and why)

Linuxbrew currently does not support brew services.  This was [confirmed on Twitter](https://twitter.com/sjackman/status/1103712101153923072).

# What happened (include command output)

There is no note in the install about the service (or how to manually start the program) at the end of install like there is in macOS ([my service](https://github.com/Homebrew/homebrew-core/pull/37634)):

```
morgo@ryzen:/tmp$ brew install tidb-server.rb 
==> Cloning https://github.com/pingcap/tidb.git
Updating /home/morgo/.cache/Homebrew/tidb-server--git
==> Checking out tag v2.1.5
HEAD is now at 4dc3093 CHANGELOG: add changelog for v2.1.5 (#9501)
HEAD is now at 4dc3093 CHANGELOG: add changelog for v2.1.5 (#9501)
==> make server
==> Caveats
This install of TiDB uses goleveldb as the storage engine instead of TiKV,
and is slower than a production install. Any benchmarks will be unreliable.

tidb-server does not bundle any clients. To use the MySQL client:
brew install mysql-client
mysql -h 127.0.0.1 -P4000 -uroot
==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/tidb-server/2.1.5: 7 files, 43.8MB, built in 37 seconds

```

Also trying to list or start services produces errors:

```
morgo@ryzen:/tmp$ brew services start tidb-server
sh: 1: list: not found
sh: 1: list: not found
Error: Failure while executing; `'' load -w /home/morgo/Library/LaunchAgents/homebrew.mxcl.tidb-server.plist` exited with 1.

morgo@ryzen:~$ brew services list
sh: 1: list: not found
sh: 1: list: not found
Name        Status  User  Plist
mysql       stopped       
tidb-server started morgo /home/morgo/Library/LaunchAgents/homebrew.mxcl.tidb-server.plist

```

# What you expected to happen

If services is not yet supported, I think it is fine to print a warning if a plist is defined.  The `plist_options :manual ` should be printed regardless though.

# Step-by-step reproduction instructions (by running `brew` commands)

Try with any Formula that uses a plist:
```
brew install mysql
brew services start mysql
```

### PR DESCRIPTION
**Pull request by morgo**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
I have one failure, but I am unsure it is related:
```
rspec ./test/cask/pkg_spec.rb:7 # Cask::Pkg#uninstall removes files and dirs referenced by the pkg
```

-----

Services is only currently supported by macOS. Attempting to use services should produce a warning for now.  It is possible it might be supported in the future.

Fixes #5847

### CODE REVIEW

File: Library/Homebrew/caveats.rb [hunk 1]
```diff
@@ -136,7 +136,18 @@ def elisp_caveats
...
-  def plist_caveats; end
+  def plist_caveats
+    s = []
+    # default to brew services not being supported.
+    # macOS will overwrite this behavior.
+    if f.plist_manual
+      s << "#{Formatter.warning("Warning:")} #{Formatter.identifier(f.full_name)} wants to install a service"\
+           ", which is currently only supported on macOS!"
+      s << "You can manually execute the service with:"
+      s << "  #{f.plist_manual}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/caveats.rb]: Use `<<~EOS` instead of the `s` array and `join` below.

File: Library/Homebrew/caveats.rb [hunk 2]
```diff
@@ -136,7 +136,18 @@ def elisp_caveats
     EOS
   end
 
-  def plist_caveats; end
+  def plist_caveats
+    s = []
+    # default to brew services not being supported.
+    # macOS will overwrite this behavior.
+    if f.plist_manual
```
Comment (MikeMcQuaid) [inline Library/Homebrew/caveats.rb]: `return unless f.plist_manual`

**Comment by morgo** [comment]
@MikeMcQuaid Thank you for the feedback.  PTAL again.

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @morgo!

**Pull request merged**